### ISSUE TITLE
gr.Launch(auth = ) UI not correct in code deployed on huggingface.com URL

### ISSUE THREAD

**Issue by dwipper** [gradio-app/gradio#5932]
### Describe the bug

I implemented the gr.Launch(auth = ) with the following code:

iface.launch(auth=login_auth, auth_message= "Enter your username and password that you received from CIMS.AI. To request a login, please email 'info@cims.ai'",share=True)

While the login form displays correct on the local browser, when deployed on a huggingface.com URL, the styling is missing and the login button doesn't work.

Please see the attached screen shots.

### Have you searched existing issues?  🔎

- [X] I have searched and found no existing issues

### Reproduction

https://huggingface.co/spaces/dwipper/NILI

### Screenshot

![image](https://github.com/gradio-app/gradio/assets/140648761/fcaf4012-8135-4e99-a852-30c7d891223d)

![image](https://github.com/gradio-app/gradio/assets/140648761/7efd4d33-baab-4cca-a75c-9a327fa49f34)

### System Info

```shell
Gradio Environment Information:
------------------------------
Operating System: Darwin
gradio version: 3.47.1
gradio_client version: 0.6.0

------------------------------------------------
gradio dependencies in your environment:

aiofiles: 23.2.1
altair: 5.1.1
fastapi: 0.103.2
ffmpy: 0.3.1
gradio-client==0.6.0 is not installed.
httpx: 0.25.0
huggingface-hub: 0.16.4
importlib-resources: 6.1.0
jinja2: 3.1.2
markupsafe: 2.1.3
matplotlib: 3.7.2
numpy: 1.25.2
orjson: 3.9.7
packaging: 23.1
pandas: 2.0.3
pillow: 10.0.0
pydantic: 1.10.13
pydub: 0.25.1
python-multipart: 0.0.6
pyyaml: 6.0.1
requests: 2.31.0
semantic-version: 2.10.0
typing-extensions: 4.5.0
uvicorn: 0.23.2
websockets: 11.0.3
authlib; extra == 'oauth' is not installed.
itsdangerous; extra == 'oauth' is not installed.


gradio_client dependencies in your environment:

fsspec: 2023.9.2
httpx: 0.25.0
huggingface-hub: 0.16.4
packaging: 23.1
requests: 2.31.0
typing-extensions: 4.5.0
websockets: 11.0.3
```

### Severity

Blocking usage of gradio

**Comment by abidlabs** [comment]
Super strange! We'll look into it thanks @dwipper

**Comment by dwipper** [comment]
@abidlabs Thanks! On a separate note, did anything ever come of the idea of providing a 'template' of blocks, rows and colums, buttons, that mimics the Interface components?

**Comment by dwipper** [comment]
@abidlabs Any ideas on what's causing the issue? We are blocked from deploying our UI publicly until it works. Any work-arounds that we could use in the meantime?

**Comment by abidlabs** [comment]
Hi @dwipper we sorry haven't had a chance to look into this just yet. Just wondering, have you tried what happens if you don't apply any custom theme. Does the problem persist with the default Gradio theme?

**Comment by dwipper** [comment]
@abidlabs 
I removed the custom theme, and still a problem. I also dumbed down the auth prompt to the basic: 

iface.launch(auth=('admin','cimsai'))

Still the same issue.

Same issue when trying it with this very simple same app (at least in my browser) Can you test and see what happens for you?:


import gradio as gr
--
  |  
  | def greet(name):
  | return "Hello " + name + "!"
  |  
  | with gr.Blocks() as demo:
  | name = gr.Textbox(label="Name")
  | output = gr.Textbox(label="Output Box")
  | greet_btn = gr.Button("Greet")
  | greet_btn.click(fn=greet, inputs=name, outputs=output, api_name="greet")
  |  
  | demo.launch(auth=('admin','test'))

**Comment by dwipper** [comment]
BTW, here's the error that is getting thrown in the browser console. Look like a path/css issue?

![image](https://github.com/gradio-app/gradio/assets/140648761/b912cbc3-ab29-4b13-a210-296ca4059f94)

**Comment by abidlabs** [comment]
Ah okay I think I may know the problem

**Comment by dwipper** [comment]
When I turn on the custom theme, this is the error message:
![image](https://github.com/gradio-app/gradio/assets/140648761/ff815dce-c3c8-47e9-bbbf-92db73cc4238)

**Comment by abidlabs** [comment]
Thanks for the diagnostics @dwipper. I've fixed this on the version of gradio installed on `main`. We probably won't do a release today, but if you want to use this in your own Space, you can add this to your requirements file:

```
https://gradio-builds.s3.amazonaws.com/66fa843699f63e72b2ed2ddf94370350bebf9769/gradio-3.50.2-py3-none-any.whl
```

You'll also want to set the `sdk_version` to `3.50.2` in the README.md file.

See this Space which does this for example: https://huggingface.co/spaces/abidlabs/test-auth2

**Comment by dwipper** [comment]
@abidlabs Thanks. I made the above changes, and correct login screen shows. If I use the hard-coded user/password like this: face.launch(auth=('admin','cims.ai')) , it works. 

But if I call an auth function, it doesn't work and throws this error:

iface.launch(auth=login_auth, auth_message= "Enter your username and password that you received from CIMS.AI. To request a login, please email 'info@cims.ai'")

![image](https://github.com/gradio-app/gradio/assets/140648761/05f2fd1b-5b14-44b9-81d3-7867ff1e5781)

**Comment by abidlabs** [comment]
Hi @dwipper I can't repro this, at least not with a simple example: https://huggingface.co/spaces/abidlabs/test-auth-3

Can you look at the logs of the Space and see what the error is? It'd be great if you can try to share a self-contained example with us.

**Comment by dwipper** [comment]
Ok. I'll put something together. BTW, when I try to click on the logs tab, it doesn't do anything....

**Comment by dwipper** [comment]
This is my Auth code

`

def log_login(username):
--
  |  
  | AIRTABLE_ENDPOINT = f'https://api.airtable.com/v0/{base_id}/{user_log_table_name}'
  |  
  | # Organize data for Airtable
  | new_fields = {
  | 'user_name': str(username),
  | }
  |  
  | data = {
  | 'fields': new_fields
  | }
  |  
  | # Post data to Airtable
  | response = requests.post(AIRTABLE_ENDPOINT, headers=HEADERS, json=data)
  |  
  | # Check for errors
  | response.raise_for_status()
  |  
  |  
  |  
  | def login_auth(username, password):
  |  
  |  
  | AIRTABLE_ENDPOINT = f'https://api.airtable.com/v0/{base_id}/{users_table_name}'
  |  
  | # Query the 'users' table to check for a match with the provided username and password
  | params = {
  | 'filterByFormula': f'AND(user_name = "{username}", password = "{password}")'
  | }
  |  
  | response = requests.get(AIRTABLE_ENDPOINT, headers=HEADERS, params=params)
  |  
  | if response.status_code == 200:
  | data = response.json()
  | if data.get('records'):
  |  
  | log_login(username)
  | global logged_in_user
  | logged_in_user = username
  |  
  | return True
  |  
  | return False

`

**Comment by dwipper** [comment]
@abidlabs never mind. I found a bug in the code......and I found the logs....thanks for all the help! Have a great weekend!

**Comment by dwipper** [comment]
Hi @abidlabs @freddyaboulton. I upgraded my spaces to Gradio 4.0.2. Unfortunately, the login styling is broken again, so it looks like the fix didn't get pushed to the new build.

**Comment by abidlabs** [comment]
Hmm it looks like there's been a regression. I'll reopen while we investigate

**Comment by abidlabs** [comment]
Hi @dwipper I actually can't reproduce this. On the latest version of gradio (4.0.2), I get this: https://huggingface.co/spaces/abidlabs/test-auth-3, you can see the login page appears fine.

Would you be able to share a simple repro with me?

**Comment by dwipper** [comment]
Hi @abidlabs . It appears the simple case works per your demo. 

That said, I reproduced it with my use case, which involves this public Space: https://huggingface.co/spaces/cimsai/NILI-Mobile which connects to this private Space: https://huggingface.co/spaces/dwipper/NILI-Mobile . When I upgraded the public space to 4.0.2, the login page displayed correctly and login worked. But when I upgraded the private space to 4.0.2, the login page on the public Space broke. I did a factory reboot after each change to the requirements.txt file.

Here's a summary of my findings with using different versions in the Spaces:

Public=3.50.2,Private=3.50.2: Works
Public=3.50.2,Private=4.0.2: Works
Public=4.02.,Private=3.50.2: Works (although after factory reset/space restart, sometimes I have to switch away from the App tab to the Files tab and back for the login screen to load properly. This seems to be an intermittent problem, as I can't always reproduce it)
Public=4.0.2,Private=4.0.2: Broken

Here are console errors in the broken state (Public=4.0.2,Private=4.0.2: Broken):
![image](https://github.com/gradio-app/gradio/assets/140648761/e51b2871-f67c-4afe-8ff0-42a44c10c823)

Here's what the console looks like when the intermittent error occurs (Public=4.02.,Private=3.50.2):
![image](https://github.com/gradio-app/gradio/assets/140648761/f1642eb7-a53d-46e4-bf38-3101ed46f4f8)

 If I switch to Files and back to the App tab:
![image](https://github.com/gradio-app/gradio/assets/140648761/2b7c8ac7-3eb4-4829-9ea8-85daed2ea7d9)


I've left the public site in the broken state (Public=4.0.2,Private=4.0.2) so that you can inspect it.

I wondering if you would recommend using the Public=3.50.2,Private=4.0.2 configuration since that seems to consistently work? My concern would be running into other issues using different versions?

If that scenario is ok for now, all good. I do have a real show stopper in figuring out how to get the username from the login in the private Sapce? Is there a way to fetch that when I run a function in the private Space?

**Comment by dwipper** [comment]
@abidlabs So I created a new Space (https://huggingface.co/spaces/dwipper/login_test), and copied the code, requirements.txt, etc. over, and the Login page is fine. It's still broken in the orginal space using 4.0.2. Not sure how that makes sense (have done factory reboots multiple times.) So I guess we can close this one (again).

**Comment by dwipper** [comment]
Humm...now when I point my public login Space to the new private Space, I get this:

```
Fetching Space from: https://huggingface.co/spaces/dwipper/NILI-Mobile
Loaded as API: https://dwipper-nili-mobile.hf.space ✔
/home/user/app/app.py:93: GradioUnusedKwargWarning: You have unused kwarg parameters in Image, please remove them: {'streamable': False, '_selectable': False}
  iface = gr.load(src="spaces",name="dwipper/NILI-Mobile",hf_token=HUGGING_FACE_TOKEN)
/home/user/app/app.py:93: GradioUnusedKwargWarning: You have unused kwarg parameters in Markdown, please remove them: {'_selectable': False}
  iface = gr.load(src="spaces",name="dwipper/NILI-Mobile",hf_token=HUGGING_FACE_TOKEN)
/home/user/app/app.py:93: GradioUnusedKwargWarning: You have unused kwarg parameters in Dropdown, please remove them: {'_selectable': False}
  iface = gr.load(src="spaces",name="dwipper/NILI-Mobile",hf_token=HUGGING_FACE_TOKEN)
/home/user/app/app.py:93: GradioUnusedKwargWarning: You have unused kwarg parameters in Textbox, please remove them: {'_selectable': False}
  iface = gr.load(src="spaces",name="dwipper/NILI-Mobile",hf_token=HUGGING_FACE_TOKEN)
/home/user/app/app.py:93: GradioUnusedKwargWarning: You have unused kwarg parameters in Dataset, please remove them: {'_selectable': False, 'component_props': [{'lines': 3, 'max_lines': 20, 'placeholder': 'Enter your question here', 'label': 'NIL Question', 'info': 'Depending the scope of your question, results may take 30-60 sec.', 'show_label': True, 'container': True, 'scale': None, 'min_width': 160, 'interactive': True, 'visible': True, 'elem_id': None, 'autofocus': False, 'autoscroll': True, 'elem_classes': [], 'type': 'text', 'text_align': None, 'rtl': False, 'show_copy_button': False}]}
  iface = gr.load(src="spaces",name="dwipper/NILI-Mobile",hf_token=HUGGING_FACE_TOKEN)
/home/user/app/app.py:93: GradioUnusedKwargWarning: You have unused kwarg parameters in Button, please remove them: {'_selectable': False}
  iface = gr.load(src="spaces",name="dwipper/NILI-Mobile",hf_token=HUGGING_FACE_TOKEN)
/home/user/app/app.py:93: GradioUnusedKwargWarning: You have unused kwarg parameters in HTML, please remove them: {'_selectable': False}
  iface = gr.load(src="spaces",name="dwipper/NILI-Mobile",hf_token=HUGGING_FACE_TOKEN)
Traceback (most recent call last):
  File "/home/user/app/app.py", line 93, in <module>
    iface = gr.load(src="spaces",name="dwipper/NILI-Mobile",hf_token=HUGGING_FACE_TOKEN)
  File "/home/user/.local/lib/python3.10/site-packages/gradio/external.py", line 70, in load
    return load_blocks_from_repo(
  File "/home/user/.local/lib/python3.10/site-packages/gradio/external.py", line 109, in load_blocks_from_repo
    blocks: gradio.Blocks = factory_methods[src](name, hf_token, alias, **kwargs)
  File "/home/user/.local/lib/python3.10/site-packages/gradio/external.py", line 505, in from_spaces
    return from_spaces_blocks(space=space_name, hf_token=hf_token)
  File "/home/user/.local/lib/python3.10/site-packages/gradio/external.py", line 511, in from_spaces_blocks
    return gradio.Blocks.from_config(client.config, predict_fns, client.src)
  File "/home/user/.local/lib/python3.10/site-packages/gradio/blocks.py", line 814, in from_config
    dependency = blocks.set_event_trigger(
TypeError: Blocks.set_event_trigger() got an unexpected keyword argument 'trigger_mode'
```

**Comment by abidlabs** [comment]
Hi @dwipper just a couple of quick points:

* How are you updating the versions of Gradio? You need to set the `sdk_version` in the `README.md` file (that takes precedence over the `requirements.txt` file)
* A Space running Gradio 4.x cannot `gr.load` a Space running Gradio 3.x, so please upgrade both Spaces to 4.x

Let me know if that helps

**Comment by dwipper** [comment]
@abidlabs Ah, I was just updating the requirements.txt and doing a factory reset.

**Comment by dwipper** [comment]
Looks like this works with the sdk_version in both enviros set to 4.1.1.
